actors/market/src/lib.rs

```diff
@@ -298,6 +297,38 @@ impl Actor {
                 }
             };
 
+            if deal.proposal.verified_deal {
+                let (mut use_client_datacap, total_client_datacap) =
+                    match all_client_datacap.get(&client_id).cloned() {
+                        None => {
+                            let total_datacap = rt
+                                .send(
+                                    &DATACAP_TOKEN_ACTOR_ADDR,
+                                    ext::datacap::BALANCE_OF_METHOD as u64,
+                                    serialize(
+                                        &Address::new_id(client_id as u64),
+                                        "balance parameters",
+                                    )?,
+                                    TokenAmount::zero(),
+                                )
+                                .and_then(|ret| datacap_balance_response(&ret))
+                                .map_err(|e| actor_error!(not_found; "failed to get datacap {}", e))
+                                .unwrap();
+
+                            (TokenAmount::zero(), total_datacap)
+                        }
+                        Some(client_data) => client_data,
+                    };
+                if use_client_datacap.clone()
+                    + TokenAmount::from_whole(deal.proposal.piece_size.0 as i64)
```

Comment:
Extract this token amount to a variable for re-use at 328. Something like `piece_datacap_required`